### DISCUSSION TITLE
Water

### DISCUSSION THREAD

**Discussion by snake-biscuits** [F1F7Y/r2radiant#59] [Ideas]
I started working on getting Water compiled into a map a little while back, but encountered a few issues
Don't really feel like digging into fixing it rn (focused on TriColl & `bsp_tool` stuff atm)

But I figured I should at least keep a record of what I learned & tried so we can come back to it later

**Comment by snake-biscuits** [comment]
This video shows how far I got:
[wet.webm](https://github.com/MRVN-Radiant/MRVN-Radiant/assets/36507175/f445d419-99f3-461b-b2f4-ca7adbeaa001)

I didn't make many edits to get here, just using flags in a `.shader`
Tho IIRC I might've messed w/ `tools/remap/source/games.cpp` to configure water as non-solid
For some reason the `CM_UNIQUE_CONTENTS` weren't updating as intended
@F1F7Y got titanclip working in the past tho

My bet is the issue comes from `TOOLS/TOOLSNODRAW` on the sides of the water brush clearing flags

The water sounds & effects were the result of 1 line of commented out C++:
[tools/remap/source/titanfall/titanfall_collisions.cpp](https://github.com/MRVN-Radiant/MRVN-Radiant/blob/main/tools/remap/source/titanfall/titanfall_collisions.cpp#LL280C1-L289C6)
```c++
int test = 0;
for (int i = 0; i < 6; i++) {
    if (axials[i] != nullptr) {
        // Titanfall::Bsp::cmBrushSideProperties.emplace_back(Titanfall::EmitTextureData(*axials[i]));
        Titanfall::Bsp::cmBrushSideProperties.emplace_back(0);  // <-- oops
    } else {
        test++;
        Titanfall::Bsp::cmBrushSideProperties.emplace_back(MASK_DISCARD);
    }
}
```
Since the water brush was the first thing I made, all surfaces defaulted to the water `$surfaceproperty`

**Reply by F1F7Y** [reply]
I plan on rewriting titanfall collision in my refactor as currently CM_UNIQUE_CONTENTS can be wrong and we dont optimize for filesize
